resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/proxy/ClientInvoker.java

```diff
@@ -117,26 +120,24 @@ protected ClientInvocation createRequest(Object[] args)
          }
       }
 
-      ClientInvocationBuilder builder = null;
+      ClientConfiguration parentConfiguration=(ClientConfiguration) target.getConfiguration();
+      ClientInvocation clientInvocation = new ClientInvocation(this.webTarget.getResteasyClient(), target.getUri(),
+    		  new ClientRequestHeaders(parentConfiguration), parentConfiguration);
       if (accepts != null)
       {
-         builder = (ClientInvocationBuilder)target.request(accepts);
+         clientInvocation.getHeaders().accept(accepts);
       }
-      else
-      {
-         builder = (ClientInvocationBuilder)target.request();
-      }
-
       for (int i = 0; i < processors.length; i++)
       {
          if (processors != null && processors[i] instanceof InvocationProcessor)
          {
             InvocationProcessor processor = (InvocationProcessor)processors[i];
-            processor.process(builder, args[i]);
+            processor.process(clientInvocation, args[i]);

```

Comment:
Want I wanted to say is that each `processor` is in charge of adding a "part" to the request so that at the end, after going through the whole `processors` list the request is fully built. 

So at first sight passing `ClientInvocationBuilder` can seems the right thing to do. Each processor will call a "setter" method on the builder and at the end the `build` method will be called to get a `ClientInvocation` ready to be invoked.

The problem is that there is no entity setter on the jaxrs `Invocation.Builder` because entity is not supposed to be "a property stored" in the builder (to my mind).   The only way to pass entity to the `Invocation.Builder` is at build phase, because entity is supposed to be a "property stored" in the built `Invocation` (to my mind) . 

So if we would have liked to properly use `ClientInvocationBuilder` in MessageBodyParameterProcessor we would have called the `build()` method. But as a common contract of `BuilderPattern.build()` it would have returned a new `ClientInvocation` and our current `ClientInvocationBuilder`  will not have been impacted since entity is not "stored" in it.

So passing `ClientInvocationBuilder` appears  not to be so good that why I guess `MessageBodyParameterProcessor` accessed `ClientInvocation` instead.

Hope it's a bit clearer.

The second and main reason why I used  `ClientInvocation` instead is that, with my changes in `ClientInvocationBuilder.build()' method, `MessageBodyParameterProcessor`  set entity in the invocation and after that, call to `ClientInvocationBuilder.build()'  reset it.